sklearn/tree/_criterion.pyx

```diff
@@ -689,6 +689,99 @@ cdef class Gini(ClassificationCriterion):
         impurity_right[0] = gini_right / self.n_outputs
 
 
+cdef class GainRatio(Entropy):
```

Comment:
I believe you still need to override the node_impurity and children_impurity using the parent's (Entropy's) ones.

Not entirely sure, but I guess along the lines:

```
    cdef double IV(self) nogil:
        cdef double iv = 0
        if self.weighted_n_left > 0:
            iv -= ((self.weighted_n_left / self.weighted_n_node_samples)
                   * log(self.weighted_n_left / self.weighted_n_node_samples))
        if self.weighted_n_right > 0:
            iv -= ((self.weighted_n_right / self.weighted_n_node_samples)
                   * log(self.weighted_n_right / self.weighted_n_node_samples))
        return iv

    cdef double node_impurity(self) nogil:
        return Entropy.node_impurity(self) / self.IV()

    cdef void children_impurity(self, double* impurity_left,
                                double* impurity_right) nogil:
        Entropy.children_impurity(self, impurity_left,
                                  impurity_right)
        cdef double iv = self.IV()
        impurity_left[0] /= iv
        impurity_right[0] /= iv
```

Reply:
Please note that this is not really compatible with what you have, what I mean is that you still need those two functions for this criterion, since they're not the same as `Entropy`, which is what is used right now with your implementation.

Reply:
Thanks for the comments. I'll have some deeper thought on that. For the sake of clarity it might be wiser, to not inherit as much as I'm doing right now. In the IV-function, do you, by chance know, if `weighted_n_left / weighted_n_node_samples` is computed once or twice? Maybe some simple performance gain, can be achieved there as well.